File: backend/Models/Follow.cs
namespace backend.Models
{
    public class Follow
    {
        public int Id { get; set; }
        public string Status { get; set; } = "pending";
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public int FollowerId { get; set; }
        public User Follower { get; set; }

        public int FollowingId { get; set; }
        public User Following { get; set; }
    }
}

File: frontend/src/types/Follow.ts
export interface Follower {
  followerId: number;
  username: string;
  profilePic: string | null;
}

export interface Following {
  followingId: number;
  username: string;
  profilePic: string | null;
}

export interface FollowRequest {
  id: number;
  followerId: number;
  username: string;
  profilePic: string | null;
  createdAt: string;
}
